src/components/IGVBrowser.js

```diff
@@ -1,59 +1,60 @@
 import _ from 'lodash/fp'
-import { Component } from 'src/libs/wrapped-components'
-import { createRef } from 'react'
-import colors from 'src/libs/colors'
-import { Ajax } from 'src/libs/ajax'
+import { useRef, useState } from 'react'
 import { div } from 'react-hyperscript-helpers'
-import igv from 'igv/dist/igv.js'
+import { centeredSpinner } from 'src/components/icons'
+import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { reportError } from 'src/libs/error'
+import * as Utils from 'src/libs/utils'
+
 
 const igvStyle = {
-  paddingTop: '10px',
-  paddingBottom: '10px',
-  margin: '8px',
+  padding: '10px 0',
+  margin: 8,
   border: `1px solid ${colors.gray[5]}`
 }
 
-export class IGVBrowser extends Component {
-  constructor(props) {
-    super(props)
-    this.containerRef = createRef()
-  }
-
-  getTrackType(filePath) {
-    const fileTypeToTrack = {
-      'bam': 'alignment',
-      'cram': 'alignment',
-      'bed': 'annotation',
-      'vcf': 'variant'
+export const IGVBrowser = ({ selectedFiles, refGenome }) => {
+  const containerRef = useRef()
+  const [loadingIgv, setLoadingIgv] = useState(true)
+
+  Utils.useOnMount(async () => {
```

Comment:
There's a warning in the console about this returning a promise. We should pull it out into a separate function and call it imperatively in `useOnMount`